Standardize error messages for file uploads

annaswims opened the pull request:
## Description of change
in preparation for department-of-veterans-affairs/va.gov-team#14002 we want to standardize the format of the error messages and give reasonable http status codes for file uploads.  We shouldn't be intentionally returning `500` for "something about the file uploaded wasn't what we wanted"


carrierwave already offered file size validation through the `size_range` method, so we can remove our `ValidateFileSize ` custom validator.
## Original issue(s)
department-of-veterans-affairs/va.gov-team#14002

## Things to know about this PR
<!--
* Are there additions to a `settings.yml` file? Do they vary by environment?
* Is there a feature flag? What is it?
* Is there some Sentry logging that was added? What alerts are relevant?
* Are there any Prometheus metrics being collected? What Grafana dashboard were they added do?
* Are there Swagger docs that were updated?
* Is there any PII concerns or questions?
-->

<!-- Please describe testing done to verify the changes or any testing planned. -->

annaswims commented on spec/request/upload_supporting_evidence_spec.rb:
```diff
@@ -9,10 +9,31 @@
         expect(JSON.parse(response.body)['data']['attributes']['guid']).to eq SupportingEvidenceAttachment.last.guid
       end
+
+      it 'returns a 422  for a malformed pdf' do
```
@Mottie 
example error details are: 
`The uploaded PDF file is invalid and cannot be read`
`The file uploaded was too large`
`You are not allowed to upload "crt" files, allowed types: pdf, png, gif, tiff, tif, jpeg, jpg, bmp, txt`

The last one will be a bigger pain to change because it comes from a gem.

LindseySaari replied:
Is there a spec for file size too large?

annaswims replied:
kind of, yes.  There's a spec that covers the size_range  range error https://github.com/department-of-veterans-affairs/vets-api/pull/5059/files#diff-e1189c09092bfde373ce490438c2dabbR38

annaswims replied:
but beyond that I'd say carrierwave specs cover the behavior of their validator.

annaswims commented:
@erluti - This ads more coverage/confidence to the carrierwave upgrade.

LindseySaari commented:
Nice! I definitely like returning more meaningful status codes and messages, but just wondering if there was a specific reason that this wasn't implemented earlier in favor of a more generic 500 status code and error message? 🤷‍♀️

LindseySaari commented on app/uploaders/profile_photo_attachment_uploader.rb:
```diff
@@ -7,7 +7,9 @@ class ProfilePhotoAttachmentUploader < CarrierWave::Uploader::Base
   include LogMetrics
 
-  MAX_FILE_SIZE = 10.megabytes
+  def self.max_file_size
```
What is your reasoning for changing this to a class method vs a constant? I was thinking that if something is static, a constant is preferred, but since you’re referencing it in the ValidateFileSize class and it's publicly exposed, maybe the class method is the better choice. 

Sorry, probably nitpicky, lots of coffee this am :)

annaswims replied:
If I'm going to return a http 413 I need a way to trigger the "file too large" error.  I'd rather override the `max_file_size` than add a >50 mb file for a spec (50 is the max elsewhere).

erluti commented on app/uploaders/validate_file_size.rb:
```diff
@@ -8,6 +8,11 @@ module ValidateFileSize
   end
 
   def validate_file_size(file)
```
this file doesn't have a spec.  I took a stab at making one in my carrierwave update PR https://github.com/department-of-veterans-affairs/vets-api/blob/2b9e53b0cc86d525fffa93c95decfe2ebf82c91d/spec/uploaders/validate_file_size_spec.rb

LindseySaari reviewed (commented):
Let me know when you're done testing with the frontend and I can re-review after Philip merges in his [PR](https://github.com/department-of-veterans-affairs/vets-api/pull/5022#issuecomment-706248397)

annaswims commented:
I'm done testing with the FE. 🎉

annaswims commented:
Gonna come back to this one and update the text of the error messages, based upon @Mottie's suggestions.  
It needs more Canadian-style "We're Sorry"...
https://design.va.gov/patterns/messaging-dictionary#access-messaging

LindseySaari commented:
> Gonna come back to this one and update the text of the error messages, based upon @Mottie's suggestions.
> It needs more Canadian-style "We're Sorry"...
> https://design.va.gov/patterns/messaging-dictionary#access-messaging

Sounds good! Ping when you're ready 👍

annaswims commented:
I've changed my mind and will do text updates in a new PR.  Currently users see "internal server error" so our error messages are still an improvement.  @LindseySaari I added a commit to remove our extra validator code for file sizes.  Previously we were only validating max, so I've changed the behavior to validate that the file body is not empty as well (min 1 byte)

edmangimelli commented on app/uploaders/supporting_evidence_attachment_uploader.rb:
```diff
@@ -2,11 +2,12 @@
 
-  MAX_FILE_SIZE = 50.megabytes
+  def size_range
+    1...50.megabytes
```
do we want this as `1..50.megabytes` (two dots) to include 50mb?

maybe even ` 1..50.1.megabytes.round`? --seems like most file managers report to the nearest 10th of a megabyte.

annaswims replied:
this would inclued 50 mb https://github.com/carrierwaveuploader/carrierwave/blob/f15f4ba71dd2c78e8df494bf65938bbaf13960f9/lib/carrierwave/uploader/file_size.rb#L35

edmangimelli replied:
```
irb(main):001:0> (1...50).max
=> 49
```
triple dots won't include the top of the range

LindseySaari commented:
Can you add to the desc of the PR as to why the validate uploader can be removed?

LindseySaari commented:
Made a few comments @annaswims

annaswims commented:
i think I've addressed all the comments.  Hopefully I didn't miss anything.

LindseySaari reviewed (approved):
Thanks for addressing the changes 👍
